neuralprocesses/tensorflow/nn.py

```diff
@@ -85,6 +85,63 @@ def ConvNd(
         )
 
 
+def UpSamplingNd(
+    dim: int,
+    size: int = 2,
+    interp_method: str = "nearest",
+    dtype=None,
+):
+    # Only set `data_format` on the GPU: there is no CPU support.
+    if len(tf.config.list_physical_devices("GPU")) > 0:
+        data_format = "channels_first"
+    else:
+        data_format = "channels_last"
+
+    # Due to inconsistent input signatures between TensorFlow UpSampling
+    # classes, we must construct conditional layer args dependent
+    # on input dim
+    layer_kwargs = dict()
+    layer_kwargs["size"] = size if dim == 1 else (size,) * dim
+    layer_kwargs["dtype"] = dtype
+    if dim == 2 or dim == 3:
```

Comment:
```suggestion
    if dim != 1:
```

If `dim == 4`, then the logic would break. (Though, admittedly, `UpSampling4D` doesn't exist in TF, but future versions could possibly support it.)